src/test/scala/chiseltest/coverage/CompilerTest.scala

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiseltest.coverage
+
+import chisel3.Module
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage}
+import chiseltest.ChiselScalatestTester
+import chiseltest.experimental.sanitizeFileName
+import firrtl.{AnnotationSeq, EmittedFirrtlCircuitAnnotation, EmittedFirrtlModuleAnnotation, EmittedVerilogCircuitAnnotation, EmittedVerilogModuleAnnotation}
+import firrtl.options.TargetDirAnnotation
+import org.scalatest.TestSuite
+
+import java.io.File
+
+/** Base trait for tests that need to compile a circuit and inspect the resulting firrtl / Verilog  */
+trait CompilerTest extends ChiselScalatestTester { this: TestSuite =>
+  protected def annos: AnnotationSeq = Seq()
+
+  protected def compile[M <: Module](gen: => M, target: String, a: AnnotationSeq = List(), ll: String = "warn"): (String, AnnotationSeq) = {
```

Comment:
Can you write a comment to define the return string-type? It appears it's any of FIRRTL or Verilog?

Reply:
Yes.